File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dictionary = defaultdict(list)
        for i in range(len(strs)):
            key = (''.join(sorted(strs[i])))
            dictionary[key].append(strs[i])
        return [dictionary[value] for value in dictionary]

File: 0049-group-anagrams/0049-group-anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    let map = new Map<string, string[]>();

    for (const str of strs) {
        let sorted = str.split('').sort().join('')

        if (!map.has(sorted)) {
            map.set(sorted, [])
        }
        map.get(sorted).push(str);
    }
    return [...map.values()]
};